### PR TITLE
Thread-safe ED Analyzers

### PR DESCRIPTION
**Pull request by ianna**
- Migrate to thread safe analyzer

### CODE REVIEW

**Comment by ianna** [comment]
@cmsbuild please test

**Comment by ianna** [comment]
@cmsbuild please test

File: Geometry/EcalTestBeam/test/ee/SurveyToTransforms.cc [hunk 1]
```diff
@@ -52,44 +21,29 @@
...
-// class decleration
-//
+
 using namespace CLHEP;
 
-class SurveyToTransforms : public edm::EDAnalyzer {
-  public:
-      SurveyToTransforms( const edm::ParameterSet& );
-      ~SurveyToTransforms();
+class SurveyToTransforms : public edm::one::EDAnalyzer<> {
```
Comment (Dr15Jones) [inline Geometry/EcalTestBeam/test/ee/SurveyToTransforms.cc]: If it is using the TFileService then you must declare a 'shared resource'. For details see

https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedAnalysisEDAnalyzer#Using_TFileService

File: Geometry/CSCGeometry/test/stubs/CSCLayerGeometryInside.cc [hunk 2]
```diff
@@ -24,44 +20,45 @@
 
 #include <string>
 
-class CSCLayerGeometryInside : public edm::EDAnalyzer {
+class CSCLayerGeometryInside : public edm::one::EDAnalyzer<> {
```
Comment (Dr15Jones) [inline Geometry/CSCGeometry/test/stubs/CSCLayerGeometryInside.cc]: This also uses the ```TFileService``` so needs to declare the shared resource.

**Comment by Dr15Jones** [comment]
-1

**Comment by ianna** [comment]
@Dr15Jones - why?

**Comment by ianna** [comment]
@cmsbuild please test

**Comment by ianna** [comment]
@cmsbuild please test

**Comment by ianna** [comment]
@cmsbuild please test

**Comment by Dr15Jones** [comment]
The modules still do not declare which resource they are dependent upon. In https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedAnalysisEDAnalyzer#Using_TFileService you'll see that you must also do
```
usesResource("TFileService");
```

**Comment by ianna** [comment]
@Dr15Jones - thanks!

**Comment by ianna** [comment]
@cmsbuild please test

**Comment by Dr15Jones** [comment]
+1

**Comment by smuzaffar** [comment]
please test
@Ianna, previous tests were running for commit 74c2d89 which is a commit before your last change. I have request to re-start the tests and it should now trigger tests based on your latests changes i.e. ee0db90

**Pull request merged**